### ISSUE TITLE
v3.0.10 with NextJS & module.css Issues

### ISSUE THREAD

**Issue by JuHwon** [tailwindlabs/tailwindcss#6899]
**What version of Tailwind CSS are you using?**

For example: v3.0.10

**What build tool (or framework if it abstracts the build tool) are you using?**

For example: postcss 8.4.5, Next.js 12.0.7

**What version of Node.js are you using?**

For example: v14.17.1

**What browser are you using?**

For example: Chrome

**What operating system are you using?**

For example: macOS

**Reproduction URL**

https://stackblitz.com/edit/nextjs-rhgora?file=src/styles/Home.module.css

**Describe your issue**

Hi i just upgraded a project to v3 and go several issues with css modules with the following error:

```
Selector "*, ::before, ::after" is not pure (pure selectors must contain at least one local class or id)
```

I managed to reproduce it. I found that when using css modules in combination with @apply.
in this project the class `translate-y-1/2` creates the error though i think there are other faults too.

**Comment by luisorbaiceta** [comment]
I am having the same issue with `v3.0.10`. In my case adding a simple `shadow-md` in the apply rule breaks the build

**Comment by ninecodes** [comment]
Same issue for me :) have to use `v3.0.9`

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

We are working on this issue as we speak. In the meantime we rolled back the bogus commit and released a `v3.0.11` version.
